Shopipy.ApiService/Program.cs

```diff
@@ -56,35 +54,17 @@
     .AddEntityFrameworkStores<AppDbContext>()
     .AddSignInManager();
 
-builder.Services.AddAuthentication(BearerTokenDefaults.AuthenticationScheme)
-    .AddBearerToken()
-    .AddGoogle(options => {
-        var config = builder.Configuration.GetSection("Authentication:Google");
-        options.ClientId = config["ClientId"]!;
-        options.ClientSecret = config["ClientSecret"]!;
-        options.Events.OnTicketReceived = async (context) => {
-            var providerKey = context.Principal?.FindFirst(ClaimTypes.NameIdentifier)?.Value;
-            var emailAddress = context.Principal?.FindFirst(ClaimTypes.Email)?.Value;
-            if (providerKey == null) throw new InvalidOperationException("nameidentifier not found");
-            if (emailAddress == null) throw new InvalidOperationException("email not found");
-            
-            var userManager = context.HttpContext.RequestServices.GetRequiredService<UserManager<User>>();
-            var user = await userManager.FindByLoginAsync(GoogleDefaults.AuthenticationScheme, providerKey);
-            if (user == null)
-            {
-                user = new User(emailAddress) {Name = "Google", Role = UserRole.SuperAdmin};
-                await userManager.CreateAsync(user);
-                await userManager.AddLoginAsync(user, new UserLoginInfo(GoogleDefaults.AuthenticationScheme, providerKey, null));
-            }
-
-            var signInManager = context.HttpContext.RequestServices.GetRequiredService<SignInManager<User>>();
-            var principal = await signInManager.CreateUserPrincipalAsync(user);
-            await context.HttpContext.SignInAsync(principal);
-            context.HandleResponse();
-        };
-
+builder.Services.AddAuthentication()
+    .AddJwtBearer(options =>
+    {
+        var key = new SymmetricSecurityKey(Encoding.UTF8.GetBytes(builder.Configuration["Authentication:Jwt:Key"]!));
+        options.TokenValidationParameters.IssuerSigningKey = key;
+        options.TokenValidationParameters.ValidAudience = builder.Configuration["Authentication:Jwt:Audience"]!;
+        options.TokenValidationParameters.ValidIssuer = builder.Configuration["Authentication:Jwt:Issuer"]!;
     });
 
+builder.Services.AddScoped<AuthService>();
```

Comment:
Why is the service scoped?

Reply:
For such simple service any lifetime would work. AddScoped is reasonable enough default choice. You would have to elaborate further.

Reply:
Also dbcontext is scoped and I should move user manager logic into the service in the future (aka never happening todo)